src/Project/BuilderCfgFile.ts

```diff
@@ -207,8 +211,13 @@ export class BuilderCfgFile extends EventEmitter implements helpers.FileSelector
     return undefined;
   }
 
-  private onBeginImport() {
-    let cfgIni = helpers.loadCfgFile(this.cfgFilePath);
+  private onBeginImport(parsedCfg: any) {
```

Comment:
Can you change this to something else than `any` ?

Reply:
I suggest not to change existing one but to add another one for your purpose

Reply:
This method is invoked by `this.on(K_BEGIN_IMPORT, this.onBeginImport);`
I think if you want to use this method, we need to revise existing code a little bit to accept from file or `parsedCfg`
Before that, I need to understand what structure is `parsedCfg`

Reply:
The structure of `parsedCfg` is like this that I mentioned above.
```
Tools{
              'one-optimize': {
                            'input_path': 'path/to/file',
                            'output_path': 'something'
                              },
              'one-codegen': {
                           'backend': 'dummy',
                           'command': 'something'
                             }
     }
```
I thought its structure seems same with `cfgIni`

Reply:
```
interface ToolType {
  [key: string]: string;
}
```
This provides free container format to provide what you want to do.
But this doesn't explictly express required items and optional items available.
That is free JSON format can be used but that make end up communication errors between developers will uncertain formats.
So, I'd like to use a explictly defined format to express attributes (or records) to lower that error.